fs/accounting/stats.go

```diff
@@ -624,11 +624,29 @@ func (s *StatsInfo) RemoveTransfer(transfer *Transfer) {
 	s.mu.Unlock()
 }
 
-// PruneTransfers makes sure there aren't too many old transfers
+// PruneAllTransfers removes all finished transfers.
+func (s *StatsInfo) PruneAllTransfers() {
+	s.mu.Lock()
+	for i := 0; i < len(s.startedTransfers); i++ {
```

Comment:
How about range iterator?

Reply:
in range iterator stop condition is not checked on each iteration, and it goes off slice boundaries when any element is removed.

Reply:
Sometimes it works for instance with maps, but I guess it's not the case here. How about https://github.com/rclone/rclone/pull/3733/files#diff-3db88cbb7dcd4ae3715a81f3543ae867R650 then?